src/Form/Fields.php

```diff
@@ -106,6 +107,52 @@ public function fill(array $input): static
 		return $this;
 	}
 
+	/**
+	 * Find a field by key/name
+	 */
+	public function findByKey(string $key): Field|FieldClass|null
+	{
+		if (str_contains($key, '+')) {
+			return $this->findByKeyRecursive($key);
+		}
+
+		return parent::findByKey($key);
+	}
+
+	/**
+	 * Find fields in nested forms recursively
+	 */
+	public function findByKeyRecursive(string $key): Field|FieldClass|null
+	{
+		$fields = $this;
+		$names  = Str::split($key, '+');
+		$index  = 0;
+		$count  = count($names);
+		$field  = null;
+
+		foreach ($names as $name) {
+			$index++;
+
+			if ($field = $fields->get($name)) {
+				if ($count !== $index) {
+					$form = $field->form();
+
+					if ($form instanceof Form === false) {
+						return null;
+					}
+
+					$fields = $form->fields();
+				}
+
+				continue;
+			}
+
+			return null;
```

Comment:
`Suggestion:` for readability

```suggestion
			$field = $fields->get($name);

			if ($field === null) {
				return null;
			}

			if ($index < $count) {
				$form = $field->form();

				if ($form instanceof Form === false) {
					return null;
				}

				$fields = $form->fields();
			}
```

I think this should do the same, but maybe I also didn't understand the code fully?

Reply:
It does the same. I just implemented it that way and added a couple inline comments.